src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/JsonDataConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+/// <summary>
+/// A configuration class for holding Json Data config values.
+/// </summary>
+public class JsonDataConfiguration
+{
+    private const string BaseDir = "data-files";
+    private const string DefaultDataFile = "test-data-azure-emissions.json";
+    private const string RegExDir = @"^[-/a-zA-Z_\d ]*$";
+    private string? dataFileLocation;
+    private string assemblyDirectory;
+
+    public const string Key = "JsonData";
+
+
+    public JsonDataConfiguration()
+    {
+        var assemblyPath = Assembly.GetExecutingAssembly().Location;
+        assemblyDirectory = Path.GetDirectoryName(assemblyPath)!;
+        DataFileLocation = DefaultDataFile;
+    }
+
+    /// <summary>
+    /// Json data file location
+    /// </summary>
+    public string DataFileLocation
```

Comment:
NIT:  prefer `getters/setters` to be defined before the constructor

Reply:
> NIT: prefer `getters/setters` to be defined before the constructor

Do you have an example?

Reply:
Got it.